### ISSUE TITLE
Add Member as "Normal" but behaves as Admin

### ISSUE THREAD

**Issue by compumatter** [wekan/wekan#3377]
## Issue

Add these issues to elsewhere:
- Snap: https://github.com/wekan/wekan-snap/issues
- LDAP: https://github.com/wekan/wekan-ldap/issues
- UCS: https://github.com/wekan/univention/issues

Other Wekan issues can be added here.

**Server Setup Information**: LAMP
* Note: Please anonymize info, and do not add to this public issue any of your Wekan board URLs, passwords, API tokens etc, do you understand?: Yes
* Did you test in newest Wekan?: Current latest version - yes
* For new Wekan install, did you configure root-url correctly so Wekan cards open correctly https://github.com/wekan/wekan/wiki/Settings ? Yes
* Wekan version: 4.56
* Operating System: Linux 20.04
* Deployment Method(snap/docker/sandstorm/mongodb bundle/source): bundle
* Http frontend if any (Caddy, Nginx, Apache, see config examples from Wekan GitHub wiki first): Nginx
* Node Version: v12.16.2
* MongoDB Version: v4.0.19
* Wekan only works on newest desktop Firefox/Chromium/Chrome/Edge/Chromium Edge and mobile Chrome. What webbrowser version are you using? Firefox and Chrome

**Problem description**:

I've created a clear video on this issue: https://www.youtube.com/watch?v=rcQw7DT13-Q

- *Explain steps how to reproduce* 
Create a board, share it with 2 users each having "normal" default permissions.  The expected behavior is that neither user should be able to archive this board since I am the owner.  In my creation one user was still given archive ability and the other was not.

**Comment by xet7** [comment]
Wow, thanks a lot for letting me know! I did not know that some normal users can see those archive and clone buttons, it's a bug.

**Comment by compumatter** [comment]
Of course. Glad you have it on your radar now.

I could understand a normal user being able to archive / hiding it “for themselves” but not also applying to my user 

Though the best approach is just removing them as a member to the board as already exists

Sincerely,

Jay
	
Jay Lepore
CompuMatter
Office: 360 299 2428
Cell: 360 202 0434
Email: jay@compumatter.com
If you are satisfied with your service, please consider providing us a social media review at https://compumatter.com/reviews
...If you are not, please let me know so I can make it better

**Comment by Samunosuke** [comment]
I would like to add a few more things which are not correct user rights for a "normal" user!

1. I think a "normal" user should not be able to delete or add List in the board or swimmlane: 
![grafik](https://user-images.githubusercontent.com/1368406/102180354-03dad400-3ea9-11eb-8b99-343a4c3f4e9c.png)

2. A "normal" user should not be able to move or reorder the lists
![grafik](https://user-images.githubusercontent.com/1368406/102180929-fa9e3700-3ea9-11eb-87a3-d6c87e991e3d.png)

3. A "normal" user should not be able to add a list to the board or swimmlane
![grafik](https://user-images.githubusercontent.com/1368406/102181060-346f3d80-3eaa-11eb-9371-20fcab89b012.png)

4. A "normal" user should not be able to delete cards which he/she did not create
![grafik](https://user-images.githubusercontent.com/1368406/102181267-8912b880-3eaa-11eb-84da-5888ec9edd68.png)

5. A "normal" user should definitely not be able to archive a board which he did not create by himself!!!
![grafik](https://user-images.githubusercontent.com/1368406/102181418-d000ae00-3eaa-11eb-8f13-6b9bb930784b.png)

6. A "normal" user should not be able to create a new swimmlane in a private board which is not owned by himself!
![grafik](https://user-images.githubusercontent.com/1368406/102181643-47ced880-3eab-11eb-81c5-19606817c24f.png)

7. A "normal" user should not be able to archive a swimmlane in a private board which is not owned by himself!
![grafik](https://user-images.githubusercontent.com/1368406/102181748-7482f000-3eab-11eb-99e6-ac9932c16cf8.png)

8. A "normal" user should not be able to move a swimmlane in a private board which is not owned by himself
![grafik](https://user-images.githubusercontent.com/1368406/102181941-b449d780-3eab-11eb-982c-e93a3a9cebb0.png)

In short a "normal" user should not be able to mess up a board which is not owned by himself!

Thank you for creating such a wonderful tool, but these are things which should get fixed soon!

**Comment by xet7** [comment]
@Samunosuke 

You are probably describing Worker role https://github.com/wekan/wekan/wiki/REST-API-Role#worker

Current normal user permissions are not so limited.

**Comment by Samunosuke** [comment]
@xet7 No, worker can just comment cards. I think of a normal user which can add, move or delete cards which are owned by himself in a private board.
A worker is not able to describe tasks, which is not what I want.
I can't see a difference between a bord-admin and a normal board user right now. I think we need something in between.

**Comment by xet7** [comment]
@Samunosuke 

You are describing something between normal user and Worker.

**Comment by Samunosuke** [comment]
@xet7  
I think the most important thing that a "normal" user (which I have added to my board) can't delete or mess up the structure of my board. What if I already have a lot of tasks in a list and a "normal" user deletes the whole list - they will be lost forever. This is possible right now, but this is not what I would expect from a normal user.
It could be handled the one way or the other. For me it would be ok if there is some user level between an "admin" or an "normal" user, but maybe it should be considered that the normal user gets limited rights.

**Comment by xet7** [comment]
@Samunosuke 

Ok, I'll try to make those limitations to normal user, and see what feedback I get.

**Comment by pgh2357** [comment]
Thank you @Samunosuke ! You are right!
@xet7 I appreciate and thank you for your efforts!
The main complexity of User Access Levels (UAL) or User Rights Management is in the nature of how different the situation of BOARDS is where:
Case 1: There is only 1 user (admin = worker)
Case 2: There are 2 or more users and they work like "Smurfs"... Grand-Smurf creates and edits cards, all other just check/uncheck the to-do-lists in cards and move cards from one list ("in progress") to another ("done" or "for testing")
Case 3: There are 2 or more users and they all have the right to create new cards, edit them, check/uncheck list-items in to-do-lists of cards and also move the cards from one list ("in progress") to another ("done" or "for testing").
All of three cases could be handled when:
1. Nobody else (except admins) would be allowed to delete a LIST with all CARDS (or empty) in a Not-Private-BOARD.
2. Even when the admin himself is going to delete the LIST... it would be much better (from the UX and security perspective) when a special dialog is shown with "Continue" and "Cancel" buttons where the user is asked "You are going to delete the LIST <name_of_list> with all cards inside! Are you really sure you want to delete the LIST ?"
A "Stop" sign/icon with some Red and Yellow colorisations of border or buttons would also be nice. This special dialog with "Stop" icon, yellow/red colorisation, "LIST" in capital letters would all signalize the admin "Alert! Alert! Alert! Re-Think! Re-Think! Re-Think!"
3. Please consider also to make a Backup of the list before deletion (if possible?)... or maybe provide an export of the list as (XML or JSON), so just in case... admin could undo the deletion by importing that list into the board.

Deletion of a LIST means LOST of:
Dates, Timestamps, Tasks, Lists, Checked-List-Items, detailed descriptions, conversations and agreements, and all other/calculated data (like duration of a state/phase)...

**Comment by xet7** [comment]
@Samunosuke  @pgh2357

Please test newest Wekan v4.60, it has restrictions added for normal users.

**Comment by pgh2357** [comment]
@xet7 Its working! Thank you very much!
Is there any task for showing total of cards in each list as cake-dia. planned?
What about import/export function?
Kindly regards

**Comment by xet7** [comment]
> @xet7 Its working! Thank you very much!
> Is there any task for showing total of cards in each list as cake-dia. planned?

Top right click your username => Change Settings => Show cards count if list contains more than => (Add number like 1) => Click Apply.

> What about import/export function?

Import:
- All Boards => Add board => Import (Trello/From previous export/CSV)

There "From previous export" means from previous board exported to Wekan JSON.

Export, when board is open:
- Click: ☰ => ⚙ => Board Settings => Export.

**Comment by Samunosuke** [comment]
@xet7 
Thank you very much! I can confirm that it is working in V 4.60.
That is exactly what we need!

**Comment by Samunosuke** [comment]
@xet7 
Thanks again! I can confirm that the "normal" user can delete check-list items and the check-list in V 4.61. 
From my perspective the issue can be closed.

**Comment by xet7** [comment]
@compumatter 

Please test Wekan v4.61 . Does your original issue still happen?

**Comment by compumatter** [comment]
A related issue was depicted in the video still exists ie;  From the creator of the board, the icons are different colors while both for 'Normal' permissions still exists ie; as seen in attached image. 
![normal-as-admin](https://user-images.githubusercontent.com/121611/102440498-2e29b000-3fd5-11eb-9499-5a75ba718dcc.jpg)

However, the main issue ie; which allows 'Normal' users to archive the entire board is now resolved!  Thank you.

**Comment by xet7** [comment]
@compumatter 

It seems that icon color is member presence status, like idle etc: 
- https://github.com/wekan/wekan/blob/master/client/components/users/userAvatar.js#L22-L30

Well, I did not know Wekan has that feature. I just found it from code.

**Comment by PizzaProgram** [comment]
> Export, when board is open:
> 
>     * Click: ☰ => gear => Board Settings => Export.

Sadly, there is no EXPORT possibility. 
(Fresh v4.7 install, first time use. Created 100 cards, do not want to loose...)
![kép](https://user-images.githubusercontent.com/26026089/104103340-d57ec780-52a1-11eb-9e41-d939226eda4b.png)

**Comment by PizzaProgram** [comment]
Also, it is not possible to Add second user to a board.
1.
![kép](https://user-images.githubusercontent.com/26026089/104103436-66ee3980-52a2-11eb-8040-5184f4a02033.png)
2.
Selecting Sandstorm member, Clicking connect...
![kép](https://user-images.githubusercontent.com/26026089/104103483-a3ba3080-52a2-11eb-8806-769d1c48e280.png)

Nothing happens.

**Comment by xet7** [comment]
@PizzaProgram 

Please move your 2 comments to new issue. In future, please do not comment closed issues. Thanks!

**Comment by xet7** [comment]
@PizzaProgram 

I moved your issue to #3423
